happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
-        # insert
-        for item in items_input:
-            item = client.create_device(device_cls=item["type"], **item)
-            exists = item["_id"] in [c["_id"] for c in client.all_items]
-            client._store(item, insert=not exists)
-    elif args.cmd == "container-registry":
-        pt = prettytable.PrettyTable()
-        pt.field_names = ["Container Name", "Container Class", "Object Class"]
-        pt.align = "l"
-        for type_, class_, in happi.containers.registry._registry.items():
-            pt.add_row([type_,
-                        f'{class_.__module__}.{class_.__name__}',
-                        class_.device_class.default])
-        print(pt)
-    elif args.cmd == "transfer":
-        logger.debug('Starting transfer block')
-        # verify name and target both exist and are valid
-        item = client.find_device(name=args.name)
-        registry = happi.containers.registry
-        # This is slow if dictionary is large
-        target_match = [k for k, _ in registry.items() if
-                        args.target in k]
-        if len(target_match) > 1 and args.name in target_match:
-            target_match = [args.name]
-        elif len(target_match) != 1:
-            print(f'Target container name ({args.target}) '
-                  'not specific enough')
-            sys.exit(1)
-        target = happi.containers.registry._registry[target_match[0]]
-        # transfer item and prompt for fixes
-        transfer_container(client, item, target)
+            try:
+                value = device._info_attrs[field].enforce_value(value)
+            except ValueError:
+                logger.error(f'Error enforcing type for value: {value}')
+                raise click.Abort()
+
+        # set field
+        try:
+            getattr(device, field)
+            logger.info(f'Setting {name}.{field}: {value}')
+            setattr(device, field, value)
+        except Exception as e:
+            logger.error(f'Could not edit {name}.{field}: {e}')
+            raise click.Abort()
+
+    device.show_info()
+    if click.confirm('Please confirm the device info is correct'):
+        logger.info('Editing device')
+        device.save()
+    else:
+        logger.info('Aborting')
+
+
+@happi_cli.command()
+@click.argument('device_names', nargs=-1)
+@click.pass_context
+def load(ctx, device_names):
+    """Open IPython terminal with DEVICE_NAMES loaded"""
+
+    logger.debug('Starting load block')
+    # retrieve client
+    client = ctx.obj
+
+    logger.info(f'Creating shell with devices {device_names}')
+    devices = {}
+    names = " ".join(device_names)
+    names = names.split()
+    if len(names) < 1:
+        raise click.Abort()
+
+    for name in names:
+        devices[name] = client.load_device(name=name)
+
+    from IPython import start_ipython  # noqa
+    start_ipython(argv=['--quick'], user_ns=devices)
+
+
+# TODO: FIgure out how to deal with json and click.  list of args doesn't
```

Comment:
What exactly doesn't work here?

Reply:
It's possible this works now, I was worried about json inputs at some point.  Maybe it works now, tests seem to pass.  I'll revisit